src/app/invest/invest.component.ts

```diff
@@ -17,63 +15,34 @@ import { CurrencyDefaultPipe } from '../shared/pipes/currency-default.pipe';
 })
 export class InvestComponent implements OnInit {
     project$: Observable<Project>;
-    projectWalletMW$: Observable<ProjectWalletInfo>;
+    investmentData$: Observable<InvestmentData>;
 
-    investmentDetails: DetailsResult;
-
-    maxInvestReached = false;
-    minUserInvest: number;
-    maxUserInvest: number;
-    totalInvestment: number;
-
-    oneYearReturn: string;
-    fiveYearsReturn: string;
-    tenYearsReturn: string;
```

Comment:
> Consider reading this review in the context of [this](https://github.com/AMPnet/ampnet-crowdfunding-frontend/pull/145/commits/fc7590276710baf6ad810b58fd4f47b2baa706ad) commit.
> 
> I've refactored things a bit because the case was a bit complicated. As I said few weeks ago: if you subscribe in components, it's very likely that you don't need it, so try to avoid them and subscribe in templates (look at the [article](https://indepth.dev/rxjs-in-angular-when-to-subscribe-rarely/) I've mentioned before).
> 
> Try to test this in your own browser if everything seems ok, and check if I made some typo. Thanks.


Thank you for this detailed review.
When I look at your code now, I see that I first started to code the same way you did with
no subscribers in component than in the template,
but then I have multiple request problems that were happening whenever I changed the amount.
I didn’t know how else to solve that problem other than to subscribe in the component.
I would greatly appreciate it if we can go through this code together,
cause I have some questions regarding "shareReplay(1)" and "take(1)" operators.

Thanks!